js/scripts.js

```diff
@@ -1 +1,137 @@
+}																							// Math.floor zaokrągla liczby zmiennoprzecinkowe w dół do całkowitej liczby (tu: 0, 1 , 2)
+																							// dzięki temu zwracana wartość funkcji to zawsze jedna z 3 opcji!
+
+var playerPickElem = document.getElementById('js-playerPick');								// Tworzymy zmienne playerPickElem, computerPickElem, playerResultElem, computerResultElem
+var computerPickElem = document.getElementById('js-computerPick');							// przypisujemy im kolejne spany (linie) z tabliczy wyników
+var playerResultElem = document.getElementById('js-playerResult');							
+var computerResultElem = document.getElementById('js-computerResult');
+
+function playerPick(playerPick) {															// funkcja wyboru dla gracza i komputera - w linii 10-12 przypisaliśmy nasłuch na buttony odpowiedzialne za wybór opcji w grze
+	var computerPick = getComputerPick();													//w związku z tym kliknięcie danej opcji powoduje odpalenie funkcji playerPick 
+	playerPickElem.innerHTML = playerPick;													//z parametrem playerPick (string) wynikającym z kodu w linii 10-12 - 'rock', 'paper', 'scissors'
+	computerPickElem.innerHTML = computerPick;
+	checkRoundWinner(playerPick, computerPick);												// 1. tworzymy zmienną computerPick, a jej wartością jest wynik funkcji losujęcej dla komputera
+}																							// 2. w tablicy wyników zmieniamy kod html, na wartość parametru playerPick(linia 12)
+																							// 3. w tablicy wyników zmieniamy kod html, na wartość wynikającą z funkcji getComputerPick (zmiennej computerPick)
+																							// 4. odpalamy funkcję sprawdzającą kto wygrał w grze checkRoundWinner z parametrami równymi zmiennym playerPick i computerPick!
+
+
+
+
+function checkRoundWinner(playerPick, computerPick) {										// Funkcja sprawdzająca wygraną w grze przyjmująca jako parametry 2 zmienne 
+	playerResultElem.innerHTML = computerResultElem.innerHTML = '';							// 1. Ustala wartość wyniku na ' ' (zmienia kod HTML)
+	var winnerIs = 'player';																// 2. Tworzy zmienną winnerIs i daje jej domyślną wartość 'player'
+	if (playerPick == computerPick) {														// 3. Poprzez analizę przypadków, porównuje wartość parametrów playerPick i ComputerPick
+		winnerIs = 'noone'; // remis														// w zależności od wyniku tej analizy przypisuje zmiennej winnerIs wartości 'computer' lub 'noone'  
+	} else if (																				// ew. nic nie zmienia (patrz punkt 2)
+		(computerPick == 'rock' &&  playerPick == 'scissors') ||
+		(computerPick == 'scissors' &&  playerPick == 'paper') ||
+		(computerPick == 'paper' &&  playerPick == 'rock')) {
+		winnerIs = 'computer';
+	}
+
+	if (winnerIs == 'player') {
+		playerResultElem.innerHTML = "Win!";
+		player.score++;
+		setGamePoints();
+	} else if (winnerIs == 'computer') {
+		computerResultElem.innerHTML = "Win!";
+		computer.score++;
+		setGamePoints();
+	}
+}
```

Comment:
Zauważ, że w tej funkcji robimy trochę masło maślane. Na podstawie picku gracza i komputera uzupełniamy zmienną `winnerIs`, a potem… sprawdzamy tą zmienną i zależnie od jej wartości podejmujemy akcje.

Spróbuj przerobić to tak żebyśmy zamiast wypełniać zmienną `winnerIs` - od razu podejmowali odpowiednie akcje. Te akcje najlepiej ubrać w nową funkcję bo w obecym stanie rzeczy ona robi zbyt wiele rzeczy na raz. Teoretycznie funkcja powinna robić maksymalnie jedną rzecz (zasada single responsibility)

Reply:
Poniżej kolejne podejście :) Nie jestem pewny czy dobrze Cię zrozumiałem. W razie co pisz to będę zmieniał dalej :)